creator-node/src/services/prometheusMonitoring/prometheusRegistry.js

```diff
@@ -1,36 +1,164 @@
+    this.addRoutesDurationTracking = this.addRoutesDurationTracking.bind(this)
+    this.addMetricName = this.addMetricName.bind(this)
+    this.addBasicHistogramMetric = this.addBasicHistogramMetric.bind(this)
+    this.getDurationTrackingMetricName =
+      this.getDurationTrackingMetricName.bind(this)
+
+    this.initMetricsFromConstantsFile()
+  }
+
+  // Instantiates PrometheusRegistry with the imported custom metrics
+  initMetricsFromConstantsFile() {
+    for (const { metricType: MetricType, metricConfig } of Object.values(
+      METRICS
+    )) {
+      // Create and register instance of MetricType, with provided metricConfig
+      const metric = new MetricType(metricConfig)
+      this.registry.registerMetric(metric)
+    }
+  }
+
+  /**
+   * Add duration and status code metrics to all routes
+   *
+   * example:
+   *  metric name:
+   *    SYNC_STATUS_WALLETPUBLICKEY_SECONDS_HISTOGRAM : sync_status_walletpublickey_seconds
+   *  metric config: {
+   *    name: 'sync_status_walletpublickey_seconds',
+   *    help: 'Duration for /sync_status_:walletpublickey',
+   *    labelNames: ['code']
+   *  }
+   *
+   * @param {Object[]} routes Array of objects that look like {path: the api route, method: the api method}
+   */
+  async addRoutesDurationTracking(routes) {
+    const addDurationTracking = async ({ path, method, regex }) => {
+      const name = this.getDurationTrackingMetricName(path, method)
+
+      try {
+        this.addBasicHistogramMetric({
+          name,
+          doc: `Duration for ${path}`
+        })
+        this.addMetricName({
+          key: `${name}_HISTOGRAM`.toUpperCase(),
+          value: name
+        })
+        this.addRouteRegex({
+          path,
+          method,
+          regex
+        })
+      } catch (e) {
+        genericLogger.warn(
+          `DurationTracking || Could not add metrics for ${path} with method ${method}: ${e.message}`
+        )
+      }
+    }
+
+    // Create metrics to track duration and status code for every route
+    for (const route of routes) {
+      await addDurationTracking(route)
+    }
+  }
+
+  /**
+   * Create a basic histogram metric and register it
+   * @param {Object} config
+   * @param {string} config.name the name of the metric
+   * @param {string} config.doc english documentation explaining metric
+   * @param {string[]} config.labels custom labels of other things to monitor; e.g. a particular cid
+   */
+  addBasicHistogramMetric({ name, doc: help, labels = [] }) {
+    // If name or help is null, or name has special characters, do not attempt to register metric
+    if (!name || !help || name.match(/[^A-Za-z_0-9]/)) {
+      throw new Error(`Improper inputs: name=${name} doc=${help}`)
+    }
+
+    const metric = new PrometheusClient.Histogram({
+      name,
+      help,
+      labelNames: ['code' /* status code of route */, ...labels],
+      buckets: [0.1, 0.3, 0.5, 1, 3, 5, 10] // 0.1 to 10 seconds
```

Comment:
can change, this was just a placeholder bucket

Reply:
i think it should be like 1,5,10,15,20,30,45,60? we definitely need longer than 10s cause some stuff can take a little bit